### PR TITLE
[Mobile Payments] Fix card reader connection modal transitions

### LINKED ISSUE #7686: [Mobile Payments] Transition issues with Several Found Readers connection modal
## Description

When connecting a card reader, and choosing between multiple readers, some of the transitions aren't quite right.

- [ ] Connection flow (`Scanning for reader`) does not animate in.
- [ ] There is a brief white flash caused by the overlay view disappearing between `Keep searching` and the `Several readers found` modal.
- [ ] The grey overlay view is a different colour between the connection modal and the `Several readers found` modal.
- [ ] After tapping `Connect` on a reader, the `Several readers found` modal is animated with a slide down off screen, then immediately replaced with the `Connecting to reader` modal.
- [ ] For some of the time `Connecting to reader` is on screen, an extra overlay is placed over the whole screen.
- [ ] When the reader connection is complete, the grey overlay is removed before the modal above it.
- [ ] The modal is not animated off at the end of the flow.

## Screenshots

https://user-images.githubusercontent.com/2472348/189180339-c43e1a92-507f-450b-adfa-adac6247f400.mp4

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #7686 

### Description

When connecting a card reader, and choosing between multiple readers, some of the transitions weren't quite right.

This PR changes the flow to use a system transition style for all modal presentation and dismissal, and corrects the `animated` flags for each transition. It doesn't resolve all the issues identified, but gets most of them without making major changes.

- [x] Connection flow (`Scanning for reader`) does not animate in.
- [ ] There is a brief white flash caused by the overlay view disappearing between `Keep searching` and the `Several readers found` modal.
- [x] The grey overlay view is a different colour between the connection modal and the `Several readers found` modal.
- [x] After tapping `Connect` on a reader, the `Several readers found` modal is animated with a slide down off screen, then immediately replaced with the `Connecting to reader` modal.
- [x] For some of the time `Connecting to reader` is on screen, an extra overlay is placed over the whole screen.
- [x] When the reader connection is complete, the grey overlay is removed before the modal above it.
- [x] The modal is not animated off at the end of the flow.

The design of the modal flow seems to prevent us from fixing the brief white flash when the overlay view disappears. Switching between modals is two operations: first dismiss, then present, and the overlay is part of the views that are dismissed and presented. A redesign could have a host view controller be responsible for the overlay, then the display and dismissal of the various modals, only dismissing the host controller when the flow ends.

The animated dismissal and replacement of the `Several readers found` screen is also a brief white flash now.

> - [x] After tapping `Connect` on a reader, the `Several readers found` modal is animated with a slide down off screen, then immediately replaced with the `Connecting to reader` modal.

### Testing instructions

You need multiple readers to test this; the simulated readers can also be used if you run it from Xcode.
Use `Edit Scheme > Arguments > -simulate-stripe-card-reader ✅` to use the simulator.

1. Turn on all your card readers
2. Navigate to `Menu > Payments`
3. Tap `Manage card reader`
4. Tap `Connect card reader`
5. Tap `Keep searching` when the first reader comes up*
6. Tap connect on a reader in the list of readers, when it shows

Repeat the above and similar steps, including cancel at different steps. Check that the transitions are more consistent than in the Before video below (they aren't perfect, just better.)

The flow can also be accessed when taking a payment without a connected reader: see `Payments > Collect payment`, `Orders > + > Create Order`, and `Orders > Order details > Collect Payment` for these flows.

*This will only come up if you've not got a saved reader which it auto-connects to. If it auto-connects, just tap `Disconnect` for that reader and try again.

#### Before 
https://user-images.githubusercontent.com/2472348/189184113-429f1f14-5a44-4b9f-8da4-572e2f7ed6d9.mp4

#### After
https://user-images.githubusercontent.com/2472348/189184595-ef94ce0c-ec6e-48e4-8949-6ebbbbf7174d.mp4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerce.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -11036,7 +11040,7 @@
 				ASSETCATALOG_COMPILER_WIDGET_BACKGROUND_COLOR_NAME = WidgetBackground;
 				CLANG_CXX_LANGUAGE_STANDARD = "gnu++17";
 				CLANG_ENABLE_OBJC_WEAK = YES;
-				CODE_SIGN_ENTITLEMENTS = StoreWidgets/Entitlements/StoreWidgets-Debug.entitlements;
+				CODE_SIGN_ENTITLEMENTS = "StoreWidgets/Entitlements/StoreWidgets-Debug.entitlements";
```
Comment (toupper) [inline WooCommerce/WooCommerce.xcodeproj/project.pbxproj]: I am wondering the reason for these changes, adding "" to the entitlements, perhaps the Xcode version?

File: WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderModalFlowViewControllerProtocol.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import Foundation
+import UIKit
+
+protocol CardReaderModalFlowViewControllerProtocol: UIViewController {
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderModalFlowViewControllerProtocol.swift:4]: Nice way of implementing that prepare function for both view controllers 👍

**Comment by toupper** [comment]
@joshheald code looks good to me 👍 
For testing we have this:

> Repeat the above and similar steps, including cancel at different steps. Check that the transitions are more consistent than in the Before video below (they aren't perfect, just better.)

But I cannot see any video, is it attached? :)

**Comment by toupper** [comment]
Thanks, @joshheald for making the videos visible for Chrome. Indeed, the transitions look much smoother now, nice job! :shipit: I added a couple of comments, but nothing important.

**Review by toupper — APPROVED** [review]

**Review by pachlava — APPROVED** [review]
This one I've tested with iPhone 13 and iPad Pro simulators (also, TIL how to simulate card readers 👍). Found no issues while switching between card readers, cancelling, initiating the connection via various flows.

Found one minor landscape mode issue, unrelated to PR I believe: #7695.

File: WooCommerce/WooCommerce.xcodeproj/project.pbxproj (hunk 1 above)
Comment (joshheald) [reply WooCommerce/WooCommerce.xcodeproj/project.pbxproj]: Nothing deliberate, so you're probably right... I'm using 13.4.1 though, which is the same version as CI, so I'm going to keep these changes. StoreWidgets is a new target which may have been added by someone else using a mismatching version of Xcode...

Comment (iamgabrielma) [reply WooCommerce/WooCommerce.xcodeproj/project.pbxproj]: Just looping in, I happen to use `13.4.1` as well and the same changes appeared on my [latest PR](https://github.com/woocommerce/woocommerce-ios/pull/7680#discussion_r966814519). I kept them after checking-in with César and from what it looks like, consistency with the rest of sign entitlements 👍

**Pull request merged**